lib/smack/SmackInstGenerator.cpp

```diff
@@ -882,4 +832,179 @@ void SmackInstGenerator::visitMemSetInst(llvm::MemSetInst& msi) {
   emit(rep->memset(msi));
 }
 
+void SmackInstGenerator::generateUnModeledCall(llvm::CallInst* ci) {
+  if (ci->getFunctionType()->getReturnType()->isVoidTy()) {
+    WARN("ignoring call to "+ci->getCalledFunction()->getName().str());
```

Comment:
Why is this `WARN` and the one below is `errs`?

Reply:
It appears errs is being used to print a message while SMACK is running to warn of unmodeled behavior. @shaobo-he wrote this so he might be able to give his reasoning.

Reply:
If there's no return type, then calling a procedure without an implementation is harmless otherwise we could have false alarms.

Reply:
Ah, so there are no intrinsics that have side-effects? In other words, you are saying that if there's no return type, then calling a procedure without an implementation is sound?

Reply:
> Ah, so there are no intrinsics that have side-effects? In other words, you are saying that if there's no return type, then calling a procedure without an implementation is sound?

There are some intrinsics like `mem.set` has side-effects but all of what I've seen are handled. Currently we handle `llvm.dbg` intrinsics using this function. If we give warnings printed on the screen, there will be a lot.